torch/nn/modules/loss.py

```diff
@@ -314,13 +314,13 @@ class GaussianNLLLoss(_Loss):
     where :attr:`eps` is used for stability. By default, the constant term of
     the loss function is omitted unless :attr:`full` is ``True``. If ``var`` is not the same
     size as ``input`` (due to a homoscedastic assumption), it must either have a final dimension
-    of 1 or have one fewer dimension (with all other sizes being the same) for correct broadcasting.
+    of 1 or have one fewer dimension (when comparing from the outermost dimension, with all other
```

Comment:
I would just break this into cases. I'm not sure the order of comparison makes it that clear. Either `var` has the same shape, has the same shape except its innermost dimension is 1, or has the same shape except it's "missing" the innermost dimension. In the last two cases the variance is assumed to be the same for each distribution (the distributions are assumed to be homoscedastic)